pandas/core/base.py

```diff
@@ -684,8 +684,14 @@ def _gotitem(self, key, ndim, subset=None):
         # with the same groupby
         kwargs = dict([(attr, getattr(self, attr))
                        for attr in self._attributes])
+
```

Comment:
minor, but can you put the comments before the try (and consolidate them into 1)